### PR TITLE
Fix tutorials to support zksync-ethers v6

### PR DESCRIPTION
**Pull request by uF4No**
# What :computer: 
* Fixing all tutorial scripts to adapt to the breaking changes introduced in `ethers` / `zksync-ethers` v6

- NFT Gated was already updated

# Why :hand:
* Most tutorial scripts are broken due to breaking changes in the new version of the libraries.

# Evidence :camera:
Screenshot 2024-02-14 at 16 02 42
Screenshot 2024-02-14 at 16 03 12

Screenshot 2024-02-14 at 16 04 21

# Notes :memo:
* Any notes/thoughts that the reviewers should know prior to reviewing the code?

### CODE REVIEW

File: docs/build/tutorials/smart-contract-development/account-abstraction/custom-aa-tutorial.md [hunk 1]
```diff
@@ -833,7 +834,7 @@ This functionality is built into the SDK.
 ## Deploy the Factory
 
 ::: tip
-Make sure you deposit funds on zkSync Era using [bridges](https://zksync.io/explore#bridges) before running the deployment script.
+Make sure you deposit funds on zkSync Era using [on of the available bridges](https://zksync.io/explore#bridges) before running the deployment script.
```
Comment (Romsters) [inline docs/build/tutorials/smart-contract-development/account-abstraction/custom-aa-tutorial.md:837]:
```suggestion
Make sure you deposit funds on zkSync Era using [one of the available bridges](https://zksync.io/explore#bridges) before running the deployment script.
```

File: docs/build/tutorials/smart-contract-development/account-abstraction/custom-aa-tutorial.md [hunk 2]
```diff
@@ -914,14 +919,16 @@ export default async function (hre: HardhatRuntimeEnvironment) {
...
 
   // deploy account owned by owner1 & owner2
   const tx = await aaFactory.deployAccount(salt, owner1.address, owner2.address);
   await tx.wait();
 
   // Getting the address of the deployed contract account
-  const abiCoder = new ethers.utils.AbiCoder();
+  // Always use the JS utility methods
+  const abiCoder = new ethers.AbiCoder();
+  // const multisigAddress = '0x5C9d4d089d25bE4e38356AAe47f880946749C1eA'
```
Comment (Romsters) [inline docs/build/tutorials/smart-contract-development/account-abstraction/custom-aa-tutorial.md:931]: Should this commented line be removed?

Comment (Romsters) [reply docs/build/tutorials/smart-contract-development/account-abstraction/custom-aa-tutorial.md:931]: Or is it for reference?

File: docs/build/tutorials/smart-contract-development/paymasters/timebased.md [hunk 3]
```diff
@@ -156,43 +170,47 @@ import { Wallet, Provider, Contract, utils } from "zksync-ethers";
...
   let wallet: Wallet;
   let deployer: Deployer;
   let userWallet: Wallet;
   let paymaster: Contract;
   let greeter: Contract;
+  let paymasterAddress: string;
 
   before(async function () {
-    const deployUrl = hardhatConfig.networks.zkSyncTestnet.url;
+    const deployUrl = hardhatConfig.networks.inMemoryNode.url;
```
Comment (Romsters) [inline docs/build/tutorials/smart-contract-development/paymasters/timebased.md:191]: Didn't we want Sepolia for paymasters?

File: docs/build/tutorials/smart-contract-development/paymasters/erc20fixed.md [hunk 4]
```diff
@@ -95,47 +106,48 @@ dotenv.config();
...
-  const provider = new Provider("https://testnet.era.zksync.dev");
+  const provider = new Provider("https://sepolia.era.zksync.dev");
   // The wallet that will deploy the token and the paymaster
   // It is assumed that this wallet already has sufficient funds on zkSync
   const wallet = new Wallet(PRIVATE_KEY);
   const deployer = new Deployer(hre, wallet);
 
   // Deploying the paymaster
-  const paymasterArtifact = await deployer.loadArtifact("ERC20fixedPaymaster");
+  const paymasterArtifact = await deployer.loadArtifact("ApprovalPaymaster");
```
[the remark this answers is not in the document]
Comment (uF4No) [reply docs/build/tutorials/smart-contract-development/paymasters/erc20fixed.md:124]: the tutorial uses the paymasters provided by the CLI templates, but it's named ApprovalPaymaster, not ERC20fixedPaymaster. It was not correct before

File: docs/build/tutorials/smart-contract-development/paymasters/timebased.md [hunk 5]
```diff
@@ -156,43 +170,47 @@ import { Wallet, Provider, Contract, utils } from "zksync-ethers";
...
 import dotenv from "dotenv";
+
 dotenv.config();
 
-const PRIVATE_KEY = process.env.WALLET_PRIVATE_KEY || "0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110";
+// test pk rich wallet from in-memory node
+const PRIVATE_KEY = "0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110";
 
-describe("TimeBasedPaymaster", function () {
+describe.only("TimeBasedPaymaster", function () {
```
[the remark this answers is not in the document]
Comment (uF4No) [reply docs/build/tutorials/smart-contract-development/paymasters/timebased.md:181]: I included it cause the CLI project includes tests and I would don't want to run them as part of the tutorial

File: docs/build/tutorials/smart-contract-development/account-abstraction/daily-spend-limit.md [hunk 6]
```diff
@@ -7,29 +7,29 @@ head:
...
 - We encourage you to read [the basics of account abstraction on zkSync Era](../../../developer-reference/account-abstraction.md) and complete the [multisig account tutorial](./custom-aa-tutorial.md) before attempting this tutorial.
 
 ::: tip Local zkSync Testing with zksync-cli
-Skip the hassle for test ETH by using `zksync-cli` for local testing. Simply execute `npx zksync-cli dev start` to initialize a local zkSync development environment, which includes local Ethereum and zkSync nodes. This method allows you to test contracts without requesting external testnet funds. Explore more in the [zksync-cli documentation](../../../tooling/zksync-cli/getting-started.md).
+Skip the hassle for test ETH by using `zksync-cli` for local testing. Simply execute `npx zksync-cli@latest dev start` to initialize a local zkSync development environment, which includes local Ethereum and zkSync nodes. This method allows you to test contracts without requesting external testnet funds. Explore more in the [zksync-cli documentation](../../../tooling/zksync-cli/getting-started.md).
```
Comment (JackHamer09) [inline docs/build/tutorials/smart-contract-development/account-abstraction/daily-spend-limit.md:24]:
```suggestion
Skip the hassle for test ETH by using `zksync-cli` for local testing. Simply execute `npx zksync-cli dev start` to initialize a local zkSync development environment, which includes local Ethereum and zkSync nodes. This method allows you to test contracts without requesting external testnet funds. Explore more in the [zksync-cli documentation](../../../tooling/zksync-cli/getting-started.md).
```

No need to use @latest. If there are any other occurrencies of that - it should also be removed.

File: docs/build/tutorials/smart-contract-development/paymasters/gasless.md [hunk 7]
```diff
@@ -35,16 +35,29 @@ Each paymaster should implement the [IPaymaster](https://github.com/matter-labs/
 
 ### Step 2 — Environment Setup
 
-Using `zksync-cli` create a new project with the required dependencies and boilerplate paymaster implementations:
+Using `zksync-cli` create a new project with the required dependencies and boilerplate paymaster implementations.
 
-`npx zksync-cli create gaslessPaymaster`
+```sh
+npx zksync-cli@latest create gaslessPaymaster
```
Comment (JackHamer09) [inline docs/build/tutorials/smart-contract-development/paymasters/gasless.md:41]:
```suggestion
npx zksync-cli create gaslessPaymaster
```

**Review by JackHamer09 — CHANGES_REQUESTED** [review]

**Review by JackHamer09 — APPROVED** [review]

**Pull request merged**